compilers/concrete-optimizer/concrete-optimizer/src/dag/unparametrized.rs

```diff
@@ -210,91 +237,114 @@ impl<'dag> DagBuilder<'dag> {
             out_shape,
             comment,
         };
-        self.add_operator(op)
+        self.add_operator(op, location)
     }
 
     pub fn add_unsafe_cast(
         &mut self,
         input: OperatorIndex,
         out_precision: Precision,
+        location: Location,
     ) -> OperatorIndex {
         let input_precision = self.dag.out_precisions[input.0];
         if input_precision == out_precision {
             return input;
         }
-        self.add_operator(Operator::UnsafeCast {
-            input,
-            out_precision,
-        })
+        self.add_operator(
+            Operator::UnsafeCast {
+                input,
+                out_precision,
+            },
+            location,
+        )
     }
 
     pub fn add_round_op(
         &mut self,
         input: OperatorIndex,
         rounded_precision: Precision,
+        location: Location,
     ) -> OperatorIndex {
         let in_precision = self.dag.out_precisions[input.0];
         assert!(rounded_precision <= in_precision);
-        self.add_operator(Operator::Round {
-            input,
-            out_precision: rounded_precision,
-        })
+        self.add_operator(
+            Operator::Round {
+                input,
+                out_precision: rounded_precision,
+            },
+            location,
+        )
     }
 
-    fn add_shift_left_lsb_to_msb_no_padding(&mut self, input: OperatorIndex) -> OperatorIndex {
+    fn add_shift_left_lsb_to_msb_no_padding(
```

Comment:
high level ops (rewritten as multi basic ops) could use the input location.

Reply:
Hmm I don't understand, isn't it what we do ? For every such operation, I write the loc as the loc of the original (compound) operation.